src/ast/context.h

```diff
@@ -49,8 +72,22 @@ class ASTContext {
   Program *root = nullptr;
 
 private:
+  // wrap potentially converts external types to internal ones. At the moment,
+  // this automatically converts the parser `location` to the `Location` class
+  // bound to the current source file.
+  template <typename T>
+  auto wrap(T &&t) -> decltype(t)
```

Comment:
JC, what else do we intend on wrapping?

Reply:
Nothing at the moment, but this is also the simplest way to automatically decay all the regular cases to the same type.